packages/address-book-controller/src/AddressBookController.ts

```diff
@@ -85,10 +69,54 @@ export type AddressBookControllerGetStateAction = ControllerGetStateAction<
   AddressBookControllerState
 >;
 
+/**
+ * The action that can be performed to list contacts from the {@link AddressBookController}.
+ */
+export type AddressBookControllerListAction = {
```

Comment:
Could you elaborate on what the purpose of these events are? They seem redundant alongside the `stateChange` action. `stateChange` already tells you exactly when address are added, removed, updated, etc.

Reply:
Hmm, OK. We could use `stateChange` to detect the same scenarios with no new events, but I can see some drawbacks with that approach. Maybe the way you've done it is best for now.